File: check-if-all-1s-are-at-least-length-k-places-away/check-if-all-1s-are-at-least-length-k-places-away.java
class Solution {
    public boolean kLengthApart(int[] nums, int k) {
        int l=0;
        boolean first=true;
        for(int i=1;i<nums.length;i++){
            if(nums[i]==0)
                l++;
            else{
                if(l<k && !first)
                    return false;
                l=0;
            }
            first=false;
        }
        return true;
    }
}

File: 1548-check-if-all-1s-are-at-least-length-k-places-away/check-if-all-1s-are-at-least-length-k-places-away.cpp
class Solution {
public:
    bool kLengthApart(vector<int>& nums, int k) {
        int n = nums.size();
        int i = 0;

        // find the first 1
        while(i < n && nums[i] == 0){
            i++;
        }

        // check all subsequent 1s
        for(int j = i + 1; j < n; j++){
            if(nums[j] == 1){
                if((j - i) <= k) return false;
                i = j;
            }
        }
        return true;
    }
};